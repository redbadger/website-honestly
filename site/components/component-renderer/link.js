// @flow
import * as React from 'react';
import styles from './styles.css';

type LinkProps = {
  children?: React.Node,
  href: string
}

export default function Link({ children, href }: LinkProps) {
  return (
    <a className={styles.a} href={href}>
      {children}
    </a>
  );
}
