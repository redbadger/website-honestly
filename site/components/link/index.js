// @flow
import React, { PropTypes } from 'react';
import { NavigationLink } from 'navigation-react';

type LinkProps = {[id: string]: any}
export default function Link(props: LinkProps) {
  const linkProps = Object.assign({}, props, {
    stateKey: props.to,
  });

  return (
    <NavigationLink
      {...linkProps}
    >
      {props.children}
    </NavigationLink>
  );
}

Link.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};
