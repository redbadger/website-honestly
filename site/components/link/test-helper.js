// @flow

import { StateNavigator, HTML5HistoryManager } from 'navigation';
import * as React from 'react';
import { NavigationHandler } from 'navigation-react';
import makeAppNavigator from '../../routes';

export function mockNavigator(parentKey?: string = 'foo') {
  // The constructor is checked in prop-types,
  // so we use it as base and then mutate

  const navigator = new StateNavigator(
    [
      { key: 'foo', route: 'foo' },
      { key: 'bar', route: 'bar' },
      { key: 'barChild', route: 'bar/child', parentKey: 'bar' },
    ],
    new HTML5HistoryManager(),
  );

  navigator.navigate(parentKey);

  return navigator;
}

type Props = {
  stateNavigator?: StateNavigator,
  children: React.Node,
};

const initStateNavigator = () => {
  const navigator = makeAppNavigator();

  navigator.navigateLink('/');

  return navigator;
};

export const Context = (props: Props) => {
  const { children, stateNavigator = initStateNavigator() } = props;

  return <NavigationHandler stateNavigator={stateNavigator}>{children}</NavigationHandler>;
};
