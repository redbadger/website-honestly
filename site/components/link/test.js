// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Link from '.';


describe('components/link', () => {
  it('renders OK children', () => {
    shallow(<Link to="foo">Hello</Link>);
  });
});
