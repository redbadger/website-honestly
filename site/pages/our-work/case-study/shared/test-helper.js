// @no-flow
import React from 'react';

import styles from './what-to-read-next/style.css';
import Link from '../../../../components/link';
import WhatToReadNextSlice from './what-to-read-next/slice';

const MockedWhatToReadNext = () => {
  const nextSlices = [
    {
      name: 'Pride in London',
      tagline: 'Delivering a best-in-class digital experience',
      image: 'pride.gif',
      linkKey: 'prideCaseStudy',
    },
    {
      name: 'Financial Times',
      tagline: 'Lasting change for a media giant',
      image: 'financialTimesReadMore.jpg',
      linkKey: 'financialTimesCaseStudy',
    },
    {
      name: 'Car Trawler',
      tagline: 'Creating complete CMS control for travel brands',
      image: 'carTrawler.jpg',
      linkKey: 'carTrawlerCaseStudy',
    },
  ];
  return (
    <div className={styles.whatNext}>
      <div className={styles.whatNext__tilesContainer}>
        {nextSlices.map((slice, index) => (
          /* disabled because list is static */
          /* eslint-disable react/no-array-index-key */
          <WhatToReadNextSlice
            key={`${slice.name}-${index}`}
            name={slice.name}
            tagline={slice.tagline}
            image={slice.image}
            linkKey={slice.linkKey}
          />
          /* eslint-enable react/no-array-index-key */
        ))}
      </div>
      <Link to="ourWorkPage" className={styles.whatNext__button}>
        See more of our work
      </Link>
    </div>
  );
};

export default MockedWhatToReadNext;
