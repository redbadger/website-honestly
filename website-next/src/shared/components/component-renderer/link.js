import React, { Component } from 'react';
import styles from './styles.css';

export default class Link extends Component {
  static propTypes = {
    children: React.PropTypes.node,
    href: React.PropTypes.string.isRequired,
  };

  render() {
    return (
      <a className={styles.a} href={this.props.href}>{this.props.children}</a>
    );
  }
}
