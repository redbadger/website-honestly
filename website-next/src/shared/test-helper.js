import TestUtils from 'react-addons-test-utils';
import equals from 'lodash/isEqual';
import ShallowTestUtils from 'react-shallow-testutils';

export function render(component) {
  const renderer = TestUtils.createRenderer();
  renderer.render(component);
  return renderer.getRenderOutput();
}

export function hasProp(type, value) {
  return (component) => component.props && component.props[type] === value;
}

export function isEqual(value) {
  return (component) => equals(value, component);
}

export function findOne(predicate, component) {
  const elem = ShallowTestUtils.findAll(component, predicate);

  if (elem.length !== 1) {
    throw new Error('Did not find exactly one match');
  }

  return elem[0];
}

export function containsOne(value, component) {
  return !!findOne(isEqual(value), component);
}

export function findWithType(type, component) {
  return ShallowTestUtils.findWithType(component, type);
}

export function findAllWithType(type, component) {
  return ShallowTestUtils.findAllWithType(component, type);
}
