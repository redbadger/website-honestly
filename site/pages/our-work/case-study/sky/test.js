import React from 'react';
import { render } from 'enzyme';

import { Context } from '../../../../components/link/test-helper';
import MockedWhatToReadNext from '../shared/test-helper';
import SkyCaseStudy from './index';

jest.mock('../shared/what-to-read-next', () => MockedWhatToReadNext);

describe('site/pages/our-work/case-study/sky/', () => {
  it('renders correctly', () => {
    expect(
      render(
        <Context>
          <SkyCaseStudy />
        </Context>,
      ),
    ).toMatchSnapshot();
  });
});
