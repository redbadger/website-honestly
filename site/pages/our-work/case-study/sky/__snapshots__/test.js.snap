// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site/pages/our-work/case-study/sky/ renders correctly 1`] = `
<div
  class="caseStudy"
>
  <div
    class="header"
  >
    <div
      class="header__container"
    >
      <div
        class="header__imageContainer"
      >
        <img
          alt="A Sky customer standing at cross-roads"
          class="header__image"
          src="header.jpg"
        />
      </div>
    </div>
  </div>
  <div
    class="content"
  >
    <div
      class="content__wrapper"
    >
      <h1
        class="content__mainTitle"
      >
        Helping Sky customers help themselves, with a refreshed customer service website
      </h1>
    </div>
    <div
      class="overview"
    >
      <div
        class="quotation quote"
      >
        <div
          class="quotation__text"
        >
          <blockquote
            class="quotation__quote"
          >
            Red Badger are recognised leaders in Agile/Lean methodologies so they could immediately start working and deliver results alongside our internal team.
          </blockquote>
        </div>
        <div
          class="quotation__author"
        >
          <div
            class="avatar__container"
            style="width:70px;height:70px;min-width:70px;min-height:70px"
          >
            <img
              alt="David Crawford"
              src="quote-author.jpg"
            />
          </div>
          <div
            class="quotation__author__wrapper"
          >
            <div
              class="quotation__author__name"
            >
              — David Crawford
            </div>
            <div
              class="quotation__author__title"
            >
              Head of Software Engineering, Sky
            </div>
          </div>
        </div>
      </div>
      <div
        class="Card listBoxContainer"
      >
        <div
          class="listBox"
        >
          <h3
            class="listBox__heading"
          >
            Results
          </h3>
          <ul>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div
                class="listBox__label"
              >
                Tripled the number of issues resolved through online self help journey
              </div>
              <div />
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div
                class="listBox__label"
              >
                Doubled the customers serviced by the advanced diagnostics capability
              </div>
              <div />
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div
                class="listBox__label"
              >
                Serviced half a million customers to date
              </div>
              <div />
            </li>
            <li
              class="listBox__element listBox__element--results listBox__item"
            >
              <div
                class="listBox__label"
              >
                50% Drop in customers pushing the 'need more help' button
              </div>
              <div />
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="content__wrapper"
    >
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Let’s make things better.
        </span>
        Delivering exceptional intuitive customer service
      </h2>
      <p
        class="content__paragraph"
      >
        Sky are continuously focused on providing customers the services they want and need. It’s part of a simple, two-pronged attack: Sky seeks to improve what it has, and provide what it doesn’t have already. As part of this, Sky found that there was a higher demand for online self-help services as customers’ own digital skills grow.
      </p>
      <p
        class="content__paragraph"
      >
        As such, Sky was determined to enhance the use of their online self-help tools for in-home diagnostics, allowing issues to be resolved quicker. Taking this, Sky set out to enable customers to troubleshoot problems for themselves, particularly during times of widespread issues. This, in order to help prevent the helpline being overwhelmed with calls for the same problem.
      </p>
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Do the right thing. Do the thing right.
        </span>
        Empowering customers to help themselves
      </h2>
      <p
        class="content__paragraph"
      >
        With a clear message of achieving customer empowerment, our onsite team at Red Badger got to work. The team focused on improving the online help and support experience by enabling customers to find content, resolve issues, navigate available services, and set up their Sky equipment correctly.
      </p>
      <p
        class="content__paragraph"
      >
        Sky was already a leader in the use of agile methodologies and lean ways of working, and were determined to work with a partner that would help them to advance these skills. To complement their established structures, augmented by Sky’s UX/Design resources, we brought in an integrated team of UX, design, and technology to deliver rapidly. Bleeding edge software was utilised to revamp the site and give it the modernisation it needed.
      </p>
      <div
        class="contactBox"
      >
        <h2
          class="contactBox__heading"
        >
          Project in mind?
        </h2>
        <button
          class="contactBox__button"
          type="button"
        >
          Tell us more
        </button>
      </div>
      <p
        class="content__paragraph"
      >
        Through the use of clustered articles, customers are now able to find what they need when they need it. Additionally, the help section was bolstered to handle spikes in activity; saving customers from having to reach for the phone. So, when the football match starts glitching, the Sky team can make sure that the solution is up on the site, or if one is still being found, can put a pop up on the page to reassure customers are that they are working to find the solution needed.
      </p>
      <p
        class="content__paragraph"
      >
        <img
          alt="Screenshot of Sky Help and Support page"
          class="articleImage"
          src="article.jpg"
        />
      </p>
      <h2
        class="content__title"
      >
        <span
          class="content__redTitle"
        >
          Creating lasting change.
        </span>
        Quick delivery with more to come
      </h2>
      <p
        class="content__paragraph"
      >
        In just five months, we delivered four major pieces of customer-facing functionality. Focused initially on in-home advanced diagnostics, the first customer facing delivery was made in less than three weeks, leaving room to implement further customer feedback.
      </p>
      <p
        class="content__paragraph"
      >
        The new diagnostic platform has had a huge effect. It tripled the number of issues resolved through the online self-help journey, it doubled the customers serviced by the advanced diagnostics, and it’s serviced half a million customers to date. 
      </p>
    </div>
  </div>
  <div
    class="whatNext"
  >
    <div
      class="whatNext__tilesContainer"
    >
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/pride"
          >
            <img
              alt="Pride in London"
              class="whatNext__image"
              src="pride.gif"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Pride in London
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/pride"
        >
          <h2
            class="whatNext__title"
          >
            Delivering a best-in-class digital experience
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/financial-times"
          >
            <img
              alt="Financial Times"
              class="whatNext__image"
              src="financialTimesReadMore.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Financial Times
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/financial-times"
        >
          <h2
            class="whatNext__title"
          >
            Lasting change for a media giant
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
      <div
        class="whatNext__tile"
      >
        <figure>
          <a
            class=""
            href="/our-work/case-study/car-trawler"
          >
            <img
              alt="Car Trawler"
              class="whatNext__image"
              src="carTrawler.jpg"
            />
          </a>
          <figcaption
            class="whatNext__caption"
          >
            Car Trawler
          </figcaption>
        </figure>
        <a
          class="whatNext__link"
          href="/our-work/case-study/car-trawler"
        >
          <h2
            class="whatNext__title"
          >
            Creating complete CMS control for travel brands
            <span
              class="whatNext__arrow_wrapper"
            >
              <svg
                class="whatNext__arrow"
                viewBox="0 0 22 22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 1l10 10-10 10-1.754-1.754 6.959-7.018H1V9.772h15.205l-6.96-7.018z"
                  fill="#FC1D42"
                  fill-rule="evenodd"
                  stroke="#FC1D42"
                />
              </svg>
            </span>
          </h2>
        </a>
      </div>
    </div>
    <a
      class="whatNext__button"
      href="/our-work"
    >
      See more of our work
    </a>
  </div>
  <section
    class="contactUsContainer"
    id="contactUs"
  >
    <h2
      class="header"
    >
      We can help you
    </h2>
    <div
      class="contentContainer"
    >
      <ul
        class="list"
      >
        <li
          class="item"
        >
          Create & validate new ideas
        </li>
        <li
          class="item"
        >
          Deliver great quality products & services, fast
        </li>
        <li
          class="item"
        >
          Be bold with technology
        </li>
        <li
          class="item"
        >
          Be more customer centric
        </li>
        <li
          class="item"
        >
          Improve efficiency with lean practices
        </li>
        <li
          class="item"
        >
          Build capability & confidence
        </li>
      </ul>
      <div
        class="imgContainer"
      />
    </div>
    <div>
      <a
        class="mailToLink"
        href="mailto:hello@red-badger.com?Subject=Can%20you%20help%20me%20with%20..."
      >
        Send an email
      </a>
      <span
        class="talkToUs"
      >
        hello@red-badger.com
      </span>
    </div>
  </section>
</div>
`;
